src/sentry/api/endpoints/organization_plugins.py

```diff
@@ -6,21 +6,50 @@
 from sentry.api.bases.organization import OrganizationEndpoint
 from sentry.api.serializers import serialize
 from sentry.api.serializers.models.organization_plugin import OrganizationPluginSerializer
-from sentry.models import Project
+from sentry.models import Project, ProjectOption
 
 
 class OrganizationPluginsEndpoint(OrganizationEndpoint):
     def get(self, request, organization):
-        _plugins = []
-
-        for project in Project.objects.filter(organization=organization):
-            for plugin in plugins.configurable_for_project(project, version=None):
-                _plugins.append(
-                    serialize(
-                        plugin,
-                        request.user,
-                        OrganizationPluginSerializer(project),
-                    )
+        # Just load all Plugins once.
+        all_plugins = dict([
+            (p.slug, p) for p in plugins.all()
+        ])
+
+        if 'plugins' in request.GET:
+            desired_plugins = set(request.GET['plugins'].split())
+        else:
+            desired_plugins = set(all_plugins.keys())
+
+        if not desired_plugins.issubset(set(all_plugins.keys())):
+            return Response({'detail': 'Invalid plugins'}, status=422)
+
+        projects = Project.objects.filter(
+            organization=organization,
+        )
+
+        # Each tuple represents an enabled Plugin (of only the ones we care
+        # about) and its corresponding Project.
+        enabled_plugins = ProjectOption.objects.filter(
+            key__in=['%s:enabled' % slug for slug in desired_plugins],
+            project_id__in=projects,
+        ).values_list('key', 'project_id')
```

Comment:
Would you be able to avoid the subquery for all projects in the organization if your conditions were

```python
ProjectOption.objects.filter(
  key__in=['%s:enabled' % slug for slug in desired_plugins],
  project__organization=organization
).prefetch_related('project')
```

I think that might let you shave down your queries even further 🏌️

Reply:
👍 Yea, a `JOIN` will probably be quicker than the `IN` here. Good call.